code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -359,16 +328,23 @@ func (sc *StorageSmartContract) newAllocationRequestInternal(
 	}
 	m.tick("add_offer")
 
-	sa.StartTime = t.CreationDate
-	sa.Tx = t.Hash
+	sa.StartTime = txn.CreationDate
+	sa.Tx = txn.Hash
 
 	// create write pool and lock tokens
-	if err = sc.createWritePool(t, sa, mintNewTokens, balances); err != nil {
+	var mld = sa.restMinLockDemand()
+	if txn.Value < mld {
```

Comment:
if it is free storage tx, this will fail, because it has 0 tokens and tokens should be minted for this pool, mintNewTokens is used for that